### PR TITLE
Ea 2190 set api setid search implemntation

### PR DESCRIPTION
**Pull request by SrishtiSingh-eu**

### CODE REVIEW

File: set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java [hunk 1]
```diff
@@ -219,13 +218,18 @@ private FindOptions buildMongoPaginationOptions(UserSetQuery query) {
 	    if(query.getItem() != null) {
 		mongoQuery.filter(WebUserSetModelFields.ITEMS + " in", query.getItem());
 	    }
+
+		if(query.getSetId() != null) {
+			mongoQuery.filter(WebUserSetModelFields.IDENTIFIER + " in",   query.getSetId());
```
Comment (gsergiu) [inline set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java:223]: the "in" operator is not needed as the setId is not a list

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java [hunk 2]
```diff
@@ -18,10 +16,10 @@
 import eu.europeana.set.definitions.model.vocabulary.WebUserSetModelFields;
 
 public class UserSetQueryBuilder extends QueryBuilder {
-    
+
     String[] fields = new String[] {WebUserSetModelFields.CREATOR, WebUserSetModelFields.VISIBILITY,
-	    WebUserSetFields.TYPE, WebUserSetFields.ITEM}; 
-    Set<String> suportedFields = Set.of(fields);
+	    WebUserSetFields.TYPE, WebUserSetFields.ITEM, WebUserSetFields.SET_ID};
+    Set<String> suportedFields = new HashSet<>(Arrays.asList(fields));
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java:22]: please revert this line of code as the java version was changes to 11

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java [hunk 3]
```diff
@@ -65,6 +63,16 @@ private UserSetQuery buildSearchQuery(Map<String, String> searchCriteria, String
 	    }
 	}
 
+	if (searchCriteria.containsKey(WebUserSetFields.SET_ID)) {
+		String setId = searchCriteria.get(WebUserSetFields.SET_ID);
+		if (setId != null && ! UserSetUtils.isSetIdNumeric(setId)) {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java:68]: the not null check is redundant Integer.parseInt already checks it

File: set-definitions/src/main/java/eu/europeana/set/definitions/model/utils/UserSetUtils.java [hunk 4]
```diff
@@ -143,5 +143,20 @@ public String fillPage(UserSet userSet, int pageIndex) {
...
    	return WebUserSetFields.DEFAULT_CREATOR_URL + userId;
     }
+
+    /**
+     * This method is to check if setId is numeric
+     *
+     * @param setId
+     * @return true if numeric
+     */
+    public static boolean isSetIdNumeric(String setId) {
```
Comment (gsergiu) [inline set-definitions/src/main/java/eu/europeana/set/definitions/model/utils/UserSetUtils.java:153]: please rename to isInteger

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java [hunk 5]
```diff
@@ -65,6 +63,16 @@ private UserSetQuery buildSearchQuery(Map<String, String> searchCriteria, String
...
+			throw new ParamValidationException(I18nConstants.INVALID_PARAM_VALUE, I18nConstants.INVALID_PARAM_VALUE,
+					new String[] { "invalid value for search field " + WebUserSetFields.SET_ID,
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java:70]: "invalid value for search field ", the information from the messages.en is sufficient

**Review by gsergiu — CHANGES_REQUESTED** [review]
please implement the required changes

File: set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java (hunk 1 above)
Comment (SrishtiSingh-eu) [reply set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java:223]: removed "in" parameter

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java (hunk 2 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java:22]: done

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java (hunk 3 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java:68]: I will remove it

File: set-definitions/src/main/java/eu/europeana/set/definitions/model/utils/UserSetUtils.java (hunk 4 above)
Comment (SrishtiSingh-eu) [reply set-definitions/src/main/java/eu/europeana/set/definitions/model/utils/UserSetUtils.java:153]: renamed

File: set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java (hunk 5 above)
Comment (SrishtiSingh-eu) [reply set-web/src/main/java/eu/europeana/set/web/search/UserSetQueryBuilder.java]: done

**Pull request merged**